### PR TITLE
Remove systemd docker container name check

### PR DESCRIPTION
**Pull request by jimmidyson**
@vishh @pwittrock Please review when you get a chance.

### CODE REVIEW

File: container/docker/factory.go [hunk 1]
```diff
@@ -273,6 +269,9 @@ func Register(factory info.MachineInfoFactory, fsInfo fs.FsInfo) error {
 				return fmt.Errorf("cAdvisor requires docker version %v or above but we have found version %v reported as \"%v\"", expected_version, dockerVersion, version_string)
 			}
 		}
+		if dockerVersion[0] == 1 && dockerVersion[1] >= 10 {
```
Comment (vishh) [inline container/docker/factory.go]: This is not always true since we can enable systemd support optionally. How do we handle that?

Comment (vishh) [inline container/docker/factory.go]: AFAIK fedora and rhel will have systemd support enabled by default.

Comment (jimmidyson) [inline container/docker/factory.go]: I run fedora & this works for me. Docker has changed cgroup hierarchy in 1.10 it seems.

Comment (vishh) [inline container/docker/factory.go]: `sudo docker daemon --exec-opt="native.cgroupdriver=systemd" -s
devicemapper` will make the daemon use systemd. I wonder if we can use
docker info to figure this out.

Comment (jimmidyson) [inline container/docker/factory.go]: I didn't know that - thanks! Will take a look in the next few days.

**Comment by jimmidyson** [comment]
@vishh This PR now simplifies stuff around systemd cgroups by ignoring the fact that systemd might be being used. I think the previous logic wasn't necessary at all - we can just limit the container name to containing the full Docker container ID format (`[a-z0-9]{64}`) & then validate the container ID via Docker inspect as we did before. This gets rid the of the need for the `nosystemd` flag too which is a good thing.

File: container/docker/factory.go [hunk 2]
```diff
@@ -46,9 +43,7 @@ var dockerRunDir = flag.String("docker_run", "/var/run/docker", "Absolute path t
 
 // Regexp that identifies docker cgroups, containers started with
 // --cgroup-parent have another prefix than 'docker'
-var dockerCgroupRegexp = regexp.MustCompile(`.+-([a-z0-9]{64})\.scope$`)
-
-var noSystemd = flag.Bool("nosystemd", false, "Explicitly disable systemd support for Docker containers")
+var dockerCgroupRegexp = regexp.MustCompile(`([a-z0-9]{64})`)
```
Comment (vishh) [inline container/docker/factory.go]: Great thought :clap: :+1:

**Comment by vishh** [comment]
LGTM.

**Pull request merged**